src/lib/support/jsontlv/README.md

```diff
@@ -0,0 +1,106 @@
+# JSON to TLV and TLV to JSON Converter
+
+## Introduction
+
+Helper functions for converting TLV-encoded data to Json format and vice versa.
+
+In order for the Json format to represent the TLV format without loss of
+information, the Json key of each element should contain the TLV element tag and
+type information.
+
+The Json key schema is as follows:
+
+```
+{ [field_name:]field_id:element_type[-sub_element_type] }
```

Comment:
So, as a consumer you would need to parse each key to isolate the:
 - optional field-name
 - field id
 - element type
 - optional element sub type

Why are those not keys of an object ? I understand that it makes the json less concise, but it also does not require the consumer to manipulate strings and have to build a custom representation to iterate the json properly. It may be some benefits that I am missing ?

Reply:
I agree that multiple implementation options are available here but the may reason behind this specific implementation choice is compactness and human friendly representation.